changelog/12324.improvement.md

```diff
@@ -0,0 +1 @@
+Add `--validate-before-training`, `--fail-on-validation-warnings` and `--validation-max-history` flags to `rasa train`.
```

Comment:
I'd add a bit more detail here to justify why this improvement was required/ requested (to align validation issues detected by rasa data validate with rasa train).